awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/AwsAppSignalsCustomizerProvider.java

```diff
@@ -57,11 +59,32 @@ public class AwsAppSignalsCustomizerProvider implements AutoConfigurationCustomi
       Logger.getLogger(AwsAppSignalsCustomizerProvider.class.getName());
 
   public void customize(AutoConfigurationCustomizer autoConfiguration) {
+    autoConfiguration.addPropertiesCustomizer(this::customizeProperties);
     autoConfiguration.addSamplerCustomizer(this::customizeSampler);
     autoConfiguration.addTracerProviderCustomizer(this::customizeTracerProviderBuilder);
     autoConfiguration.addSpanExporterCustomizer(this::customizeSpanExporter);
   }
 
+  private Map<String, String> customizeProperties(ConfigProperties configProps) {
+    Map<String, String> overrides = new HashMap<>();
+    if (isSmpEnabled(configProps)) {
+      // disable logs exporter by default
+      if (System.getProperty("otel.logs.exporter", System.getenv("OTEL_LOGS_EXPORTER")) == null) {
+        overrides.put("otel.logs.exporter", "none");
+      }
+
+      String traceEndpoint =
+          System.getProperty(
+              "otel.exporter.otlp.traces.endpoint",
+              System.getenv("OTEL_EXPORTER_OTLP_TRACES_ENDPOINT"));
+      // using grpc if trace endpoint is :4315
+      if (traceEndpoint != null && traceEndpoint.contains(":4315")) {
+        overrides.put("otel.exporter.otlp.traces.protocol", "grpc");
```

Comment:
I don't think this assumption is appropriate.  The specification provides a mechanism for [specifying the protocol to use](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#specify-protocol) independent of the endpoint.  A customer may have reason to utilize HTTP/JSON over port 4315 and this weird opaque rejection of their intent is not good.

Reply:
If SMP flag is enabled, it means customer uses CWAgent for receiving AppSignals metrics. The AppSignals and CWAgent convention is using 4315 for grpc, 4316 for http. These 2 ports are already used and hardcoded by CWAgent.
So, if both SMP is enabled and SDK uses endpoint 4315, it must be grpc protocol.